distribution/tools/java-version-checker/src/unsupportedJdkVersionEntrypoint/java/org/elasticsearch/tools/java_version_checker/JavaVersionChecker.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.tools.java_version_checker;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+/**
+ * Java 7 compatible main which exits with an error.
+ */
+final class JavaVersionChecker {
+
+    private JavaVersionChecker() {}
+
+    public static void main(final String[] args) {
+        // no leniency!
+        if (args.length != 0) {
```

Comment:
Should we even bother checking the arguments here?

Reply:
I just left this like in the old implementation. No leniency!